api/storage_service.cc

```diff
@@ -773,10 +773,15 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         return map_keys(ctx.db.local().get_keyspaces());
     });
 
-    ss::update_snitch.set(r, [](std::unique_ptr<request> req) {
+    ss::update_snitch.set(r, [&g](std::unique_ptr<request> req) {
         auto ep_snitch_class_name = req->get_query_param("ep_snitch_class_name");
-        return locator::i_endpoint_snitch::reset_snitch(ep_snitch_class_name).then([] {
-            return make_ready_future<json::json_return_type>(json_void());
+        return locator::i_endpoint_snitch::reset_snitch(ep_snitch_class_name).then([&g] {
+            auto name = locator::i_endpoint_snitch::get_local_snitch_ptr()->get_name();
```

Comment:
Let's capture the `ep_snitch_class_name`. The `get_local_snitch_ptr()` is another global that's going to be eliminated anyway.

Reply:
ep_snitch_class_name can change by reser_snitch